angular/src/app/datacart/treetable/treetable.component.ts

```diff
@@ -323,31 +331,50 @@ export class TreetableComponent implements OnInit, AfterViewInit {
      * @param event - change event
      */
     cartChanged(which) {
```

Comment:
The functions that update the tree's selection status are:
* `propagateSelectionDown(node: TreeNode, selected: boolean)` -- this marks a node in the tree as selected (or unselected) along with all its child nodes.
* `propagateSelectionUp(node: TreeNode, selected: boolean)` -- this updates the selection status for a node's ancestor nodes (based on whether something below is being selected or unselected).